packages/flutter/lib/src/material/tooltip_theme.dart

```diff
@@ -250,3 +271,18 @@ class TooltipTheme extends InheritedTheme {
   @override
   bool updateShouldNotify(TooltipTheme oldWidget) => data != oldWidget.data;
 }
+
+/// The mode that will decide when the tooltip will trigger
+enum TooltipTriggerMode {
+  /// Tooltip will not be shown automatically, but can be done manually
+  /// by calling `ensureTooltipVisible`
+  Manual,
+
+  /// Tooltip will be shown automatically after a long press.
+  /// See `GestureDetector.onLongPress` for more details.
+  LongHold,
+
+  /// Tooltip will be shown automatically after a single tap
+  /// See `GestureDetector.onTap` for more details
+  Tap,
+}
```

Comment:
Nit: newline at end of file

Reply:
Still needed.

Reply:
yeah there seems to be some mess with Android Studio+ git + github (not sure which one / if only one).
I believe it's fixed now, but locally I have 2 new lines (and no git changes).